src/plays/2048/Game2048Components/Grid.js

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect } from "react";
+import Row from "./Row";
+import TransformGrid from "../transformGrid";
+import GameLogic from "../gameLogic";
+
+function Grid(props) {
+  useEffect(() => {
+    const handleKeyDown = e => {
+      e.preventDefault();
+
+      if (props.status !== "running" && props.status !== "continue") return;
+      let result = {};
+
+      switch (e.key) {
+        case "ArrowUp":
+          result = GameLogic.addCell(TransformGrid.shiftUp(props.grid), props);
+          break;
+        case "ArrowRight":
+          result = GameLogic.addCell(
+            TransformGrid.shiftRight(props.grid),
+            props
+          );
+          break;
+        case "ArrowDown":
+          result = GameLogic.addCell(
+            TransformGrid.shiftDown(props.grid),
+            props
+          );
+          break;
+        case "ArrowLeft":
+          result = GameLogic.addCell(
+            TransformGrid.shiftLeft(props.grid),
```

Comment:
@Shivam-Katare, Extract `handleKeyDown` logic in a function and call that function in the useEffect.